tests/test_generators.py

```diff
@@ -0,0 +1,329 @@
+import os
+import unittest
+from unittest import mock
+
+import dataprofiler as dp
+import numpy as np
+import pandas as pd
+
+from synthetic_data import Generator
+from synthetic_data.generators import (
+    GraphGenerator,
+    TabularGenerator,
+    UnstructuredGenerator,
+)
+
+test_dir = os.path.dirname(os.path.realpath(__file__))
+
+
+class TestTabularGenerator(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.profile_options = dp.ProfilerOptions()
+        cls.profile_options.set(
+            {
+                "data_labeler.is_enabled": False,
+                "correlation.is_enabled": True,
+                "multiprocess.is_enabled": False,
+            }
+        )
+        dp.set_seed(0)
+
+        # create dataset and profile for tabular
+        cls.tab_data = dp.Data(os.path.join(test_dir, "data/iris.csv"))
+        cls.tab_profile = dp.Profiler(
+            cls.tab_data, profiler_type="structured", options=cls.profile_options
+        )
+        cls.uncorrelated_generator = Generator(
+            profile=cls.tab_profile, is_correlated=False, seed=42
+        )
+
+    def test_synthesize_tabular(self):
+        generator = Generator(profile=self.tab_profile, seed=42)
+        self.assertIsInstance(generator, TabularGenerator)
+        synthetic_data = generator.synthesize(100)
+        self.assertEqual(len(synthetic_data), 100)
+
+        generator = Generator(data=self.tab_data, seed=42)
+        synthetic_data_2 = generator.synthesize(100)
+        self.assertEqual(len(synthetic_data_2), 100)
+
+        # asserts that both  methods create the same results
+        # if this ever fails may need to start setting seeds
+        np.testing.assert_array_equal(synthetic_data, synthetic_data_2)
+
+    def test_is_not_correlated(self):
+        self.assertFalse(self.uncorrelated_generator.is_correlated)
+        synthetic_data = self.uncorrelated_generator.synthesize(100)
+        self.assertEqual(len(synthetic_data), 100)
+
+        # test is not correlated when starting from data
+        generator = Generator(
+            data=self.tab_data.data[:10], is_correlated=False, seed=42
+        )
+        self.assertFalse(self.uncorrelated_generator.is_correlated)
+        synthetic_data = self.uncorrelated_generator.synthesize(100)
+        self.assertEqual(len(synthetic_data), 100)
+
+    @mock.patch("synthetic_data.generators.make_data_from_report")
+    def test_synthesize_correlated_method(self, mock_make_data):
+        profile = dp.Profiler(
+            data=self.tab_data,
+            options=self.profile_options,
+            samples_per_update=len(self.tab_data),
+        )
+        instance = TabularGenerator(profile)
+        instance.method = "correlated"
+        instance.synthesize(num_samples=10)
+        mock_make_data.assert_called_once()
+
+    @mock.patch("synthetic_data.generators.generate_dataset")
+    def test_uncorrelated_synthesize_columns_to_generate(self, mock_generate_dataset):
```

Comment:
This test checks if the expected columns_to_generate hashmap was generated correctly. Checks every item to the expected and checks if the actual hashmap is a subset of expected